File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        long long maxSum = INT_MIN;
        long long curr = 0;
        for(int i=0;i<nums.size();i++){
            curr += nums[i];
            if (curr > maxSum) maxSum = curr;
            if(curr < 0) curr = 0;
        }
        return maxSum;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSum = max(nums)
        currSum = 0
        for n in nums:
            if currSum < 0:
                currSum = 0
            currSum += n
            if currSum > maxSum:
                maxSum = currSum
        return maxSum